src/components/ResultsTable.tsx

```diff
@@ -0,0 +1,526 @@
+          const { mode: view, value: viewValue } = viewsByColumn[k] || {};
+          return (
+            <td
+              style={{
+                textOverflow: "ellipsis",
+                overflow: "hidden",
+              }}
+              key={k}
+            >
+              {val === "" ? (
+                <i>[block is blank]</i>
+              ) : view === "link" || view === "alias" ? (
+                <a
+                  className={"rm-page-ref"}
+                  data-link-title={getPageTitleByPageUid(uid) || ""}
+                  href={getRoamUrl(uid)}
+                  onMouseDown={(e) => {
+                    if (e.shiftKey) {
+                      openBlockInSidebar(uid);
+                      e.preventDefault();
+                      e.stopPropagation();
+                    } else if (e.ctrlKey) {
+                      ctrlClick?.({
+                        text: toCellValue(val),
+                        uid,
+                      });
+                      e.preventDefault();
+                      e.stopPropagation();
+                    }
+                  }}
+                  onClick={(e) => {
+                    if (e.shiftKey || e.ctrlKey) {
+                      e.preventDefault();
+                      e.stopPropagation();
+                    }
+                  }}
+                  onContextMenu={(e) => {
+                    if (e.ctrlKey) {
+                      e.preventDefault();
+                      e.stopPropagation();
+                    }
+                  }}
+                >
+                  {view === "alias" ? viewValue : cell(k)}
+                </a>
+              ) : view === "embed" ? (
+                <CellEmbed uid={uid} />
+              ) : (
+                cell(k)
+              )}
+            </td>
+          );
+        })}
+        {extraColumn && <td>{extraColumn.row(r)}</td>}
+      </tr>
+    </>
+  );
+};
+
+const ResultsTable = ({
+  columns,
+  results,
+  parentUid,
+  activeSort,
+  setActiveSort,
+  filters,
+  setFilters,
+  preventSavingSettings,
+  onRefresh,
+  views,
+  pageSize,
+  setPageSize,
+  pageSizeTimeoutRef,
+  page,
+  setPage,
+  allResultsLength,
+  extraColumn,
+}: {
+  columns: string[];
+  results: Result[];
+  parentUid: string;
+  // TODO - can a lot of these settings become layout settings instead of global?
+  activeSort: Sorts;
+  setActiveSort: (s: Sorts) => void;
+  filters: FilterData;
+  setFilters: (f: FilterData) => void;
+  preventSavingSettings?: boolean;
+  views: Views;
+  pageSize: number;
+  setPageSize: (p: number) => void;
+  pageSizeTimeoutRef: React.MutableRefObject<number>;
+  page: number;
+  setPage: (p: number) => void;
+  onRefresh: () => void;
+  allResultsLength: number;
+  // @deprecated - move to `-action` columns through selections
+  extraColumn?: {
+    reserved: RegExp[];
+    width: number;
+    row: (e: Result) => React.ReactNode;
+    header: React.ReactNode;
+  };
```

Comment:
I think over time, we need to decide which of these settings are actually for all layouts, and which ones are specific to table